setup.py

```diff
@@ -11,9 +11,20 @@
 SRC_DIR = os.path.join(THIS_DIR, "src")
 SNOWPARK_SRC_DIR = os.path.join(SRC_DIR, "snowflake", "snowpark")
 CONNECTOR_DEPENDENCY_VERSION = ">=3.2.0, <4.0.0"
+INSTALL_REQ_LIST = [
+    "setuptools>=40.6.0",
+    "wheel",
+    "cloudpickle>=1.6.0,<=2.0.0",
+    f"snowflake-connector-python{CONNECTOR_DEPENDENCY_VERSION}",
+    "pyyaml",
+]
 REQUIRED_PYTHON_VERSION = ">=3.8, <3.11"
 if os.getenv("SNOWFLAKE_IS_PYTHON_RUNTIME_TEST", False):
     REQUIRED_PYTHON_VERSION = ">=3.8"
+    # replace the line with condition py < 3.11
+    INSTALL_REQ_LIST[2] = "cloudpickle>=1.6.0,<=2.0.0;python_version<'3.11'"
```

Comment:
I wonder if it makes sense to make a `CLOUDPICKLE_REQ_LIST`, and we use the current if clause to determine the value. After we determine the value of `CLOUDPICKLE_REQ_LIST`, we then add the list into `INSTALL_REQ_LIST`. This way we don't need to keep a magic number 2 here.

Reply:
I second this.

Reply:
good idea, I updated the PR, let me know how it looks